src/elements/content-sidebar/activity-feed/comment/CommentMenu.js

```diff
@@ -0,0 +1,71 @@
+// @flow
+import * as React from 'react';
+import { injectIntl, FormattedMessage } from 'react-intl';
+import getProp from 'lodash/get';
+
+import IconTrash from '../../../../icons/general/IconTrash';
+import IconEllipsis from '../../../../icons/general/IconEllipsis';
+import IconPencil from '../../../../icons/general/IconPencil';
+
+import PlainButton from '../../../../components/plain-button';
+import DropdownMenu from '../../../../components/dropdown-menu';
+import { Menu, MenuItem } from '../../../../components/menu';
+
+import messages from '../../../common/messages';
+import deleteMessages from '../inline-delete/messages';
+import { ACTIVITY_TARGETS } from '../../../common/interactionTargets';
+import { fours, nines } from '../../../../styles/variables';
+import { COMMENT_TYPE_DEFAULT, COMMENT_TYPE_TASK } from '../../../../constants';
+
+type Props = {
+    id: string,
+    isDisabled: boolean,
+    onDeleteClick?: Function,
+    onEditClick?: Function,
+    permissions?: BoxItemPermission,
+    type: typeof COMMENT_TYPE_DEFAULT | typeof COMMENT_TYPE_TASK,
+} & InjectIntlProvidedProps;
+
+const CommentMenu = (props: Props) => {
+    const { intl, isDisabled, onDeleteClick, onEditClick, permissions, type } = props;
+    const canDelete = getProp(permissions, 'can_delete', false);
+    const canEdit = getProp(permissions, 'can_edit', false);
+    const isTaskComment = type === COMMENT_TYPE_TASK;
+    const editLabel = isTaskComment ? messages.taskEditMenuItem : messages.editLabel;
+    const deleteLabel = isTaskComment ? messages.taskDeleteMenuItem : deleteMessages.deleteLabel;
+
+    return (
+        <DropdownMenu className="bcs-comment-menu-container" constrainToScrollParent isRightAligned>
```

Comment:
Noted. we used the classes matching the existing convention so this wouldn't clash with those ones. this might get separated from here, so that'd be a good time to SUIT up.